keps/sig-api-machinery/3285-subresource-finalizers/README.md

```diff
@@ -0,0 +1,812 @@
+
+### Goals
+
+- Provide a way to set a different RBAC to `finalizers` from its entire resource
+- Provide a way to audit and update only `metadata.finalizers` field
+
+### Non-Goals
+
+- Provide a way to set a different RBAC to other than `finalizers` from its entire resource
+- Provide a way to allow updating its entire resource  but disallow updating `finalizers` field
+
+## Proposal
+
+A new REST endpoint `<resource>/finalizers` is added automatically to all resources as an alternative way to access the `metadata.finalizers` fielld, which allows separate RBAC.
+The existing method to acceess `metadata.finalizers` field (through the main `<resource>` endpoint) remains unchanged.
+
+### User Stories (Optional)
+
+#### Story 1
+
+There is a controller that only updates `status` field and `finalizers` field of a CRD. Administrators of the controller can allow the controller to update only these particular fields.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+- Name collision of the endpoints is needed to be considered.
+- Other existing fields may require the same alternative way to access.
+For example, [candidates](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go) in metadata fields are:
+  - `labels`
+  - `annotations`
+  - `ownerReferences`
+  - `status/conditions`
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+Risks:
+- Bugs could make unintended fields accessible.
+
+Mitigations:
+- The new endpoint needs to be negative-tested to prove that the other fields are NOT modifiable.
+
+## Design Details
+
+Add `<resource>/finalizers` endpoint to all resources and allow access to `metadata.finalizers` for the resources through the endpoints.
+The actual field in each resource isn't added, but the request to the endpoint is handled as the operation to `metadata.finalizers`.
+`Role` and `ClusterRole` can be set to the endpoint, therefore users can set a separate RBAC to `<resource>/finalizers`.
+A PoC implementation can be found [here](https://github.com/mkimuram/kubernetes/commit/751a67e03d62397c1d21947e9e2e3a81f279a29f) (Please also see [here](https://github.com/kubernetes/enhancements/pull/2840#discussion_r723464012) for how it works. Note that API definition in this KEP will differ from the one in the PoC).
+
+```
+<<[UNRESOLVED Endpoint name]>>
+Endpoint name for the subresource is still being discussed.
```

Comment:
nit if you edit more: add the proposed options, e.g

* /finalize: already used by ...
* /finalizers: some debate about whether sub-resources should be nouns of verbs

Reply:
Added.